### PR TITLE
Migrate to nextcord

### PR DESCRIPTION
**Pull request by benjitusk**

### CODE REVIEW

**Review by DenverCoder1 — APPROVED** [review]
Thanks for your help! Looks good to me.

**Review by abrahammurciano — APPROVED** [review]

File: bot.py [hunk 1]
```diff
@@ -16,16 +15,14 @@ async def main():
...
 	intents.members = True
 
-	activity = discord.Game("with students' patience")
+	activity = nextcord.Game("with students' patience")
 
 	# empty space effectively disables prefix since discord strips trailing spaces
-	bot = commands.Bot(" ", intents=intents, activity=activity)
-
-	setattr(bot, "slash", SlashCommand(bot, override_type=True, sync_commands=True))
+	bot = commands.Bot(intents=intents, activity=activity)
```
[the remark this answers is not in the document]
Comment (DenverCoder1) [reply bot.py:24]: Currently Cogs are cog listeners only supported by the commands extension, but there has been a proposal to move cogs to the main lib for use with `nextcord.Client`. So as of now, I think `commands.Bot` with no prefix is the best option until changes are made in nextcord.

File: modules/calendar/calendar_embedder.py [hunk 2]
```diff
@@ -74,17 +74,17 @@ async def embed_event_pages(
 				# wait for author to respond to go to next page
 				await wait_for_reaction(
 					bot=self.bot,
-					message=ctx.message,
+					message=message,
 					emoji_list=[next_emoji],
-					allowed_users=[ctx.author],
+					allowed_users=[interaction.user] if interaction.user else None,  # type: ignore
```
[the remark this answers is not in the document]
Comment (DenverCoder1) [reply modules/calendar/calendar_embedder.py:79]: Yeah, I suppose asserting Member type would be better. I was unsure if User type should be accepted or not even though the function accepts Collection[Member].

File: modules/join/cog.py [hunk 3]
```diff
@@ -17,68 +13,39 @@ def __init__(self, bot: commands.Bot):
...
 	):
-		# TODO: campus and year should really take integer options, but mobile has a bug
-		await ctx.defer()
+		"""Join command to get new users' information and place them in the right roles.
+		
+		Args:
+			first_name (str): Your first name.
+			last_name (str): Your last name.
+			campus (str): Your campus (Lev or Tal).
+			year (int): Your year.
```
[the remark this answers is not in the document]
Comment (DenverCoder1) [reply modules/join/cog.py:35]: Yeah, that's one reason I think types do not belong in docstrings. It's easier to modify the types in one place than two,

[the remark this answers is not in the document]
Comment (abrahammurciano) [reply modules/join/cog.py]: If you read Google's docstring style it actually says not to put the type in the docstring if you're using type hints. The vscode extension which generates them just puts them in regardless though...

[the remark this answers is not in the document]
Comment (DenverCoder1) [reply modules/join/cog.py]: I removed the types from these. I think we should have a separate PR that goes through all the docstrings to make them consistent.

[the remark this answers is not in the document]
Comment (abrahammurciano) [reply modules/join/cog.py]: Did you check if the mobile bug still exists?

[the remark this answers is not in the document]
Comment (DenverCoder1) [reply modules/join/cog.py]: Just tested and int choices seem to go through just fine now

**Comment by DenverCoder1** [comment]
Should be good now. @abrahammurciano do you approve the latest commits?

**Review by abrahammurciano — APPROVED** [review]

**Pull request merged**